src/app/src/main/kotlin/org/vaccineimpact/api/app/controllers/NewStyleOneTimeLinkController.kt

```diff
@@ -0,0 +1,50 @@
+package org.vaccineimpact.api.app.controllers
+
+import org.vaccineimpact.api.app.RedirectValidator
+import org.vaccineimpact.api.app.app_start.Controller
+import org.vaccineimpact.api.app.context.ActionContext
+import org.vaccineimpact.api.app.repositories.Repositories
+import org.vaccineimpact.api.app.repositories.TokenRepository
+import org.vaccineimpact.api.models.helpers.OneTimeAction
+import org.vaccineimpact.api.security.WebTokenHelper
+import java.time.Duration
+
+class NewStyleOneTimeLinkController(
+        context: ActionContext,
+        private val repo: TokenRepository,
+        private val tokenHelper: WebTokenHelper,
+        private val redirectValidator: RedirectValidator = RedirectValidator())
+    : Controller(context)
+{
+    constructor(context: ActionContext, repositories: Repositories, tokenHelper: WebTokenHelper)
+            : this(context, repositories.token, tokenHelper)
+
+    val serializer = context.serializer
+
+    fun getTokenForDemographicData(): String
+    {
+        return getOneTimeLinkToken(context, repo, OneTimeAction.DEMOGRAPHY)
+    }
+
+    private fun getOneTimeLinkToken(
```

Comment:
This duplicates the existing logic exactly.

Reply:
Can we have a new test to make sure this logic is covered when we remove the old method?